### ISSUE TITLE
Error when parsing expression.

### ISSUE THREAD

**Issue by Drake53** [benjamin-hodgson/Pidgin#93]
I'm trying to use this library to implement a parser for JASS: http://jass.sourceforge.net/doc/bnf.shtml
The expression parser I wrote works in most cases, but when I tried to parse larger strings (statements and functions), I found in some cases the expression parser will give an error. After some testing I was able to reproduce the error for a simple expression:
```
(foo() > 0)
(0 > foo())
```
So, I get an error for a binary expression inside a paranthesized expression, where one of the operands is a function invocation expression.
Originally I thought the cause might be that the invocation expression parser was not part of the operator table (in JASS you can't invoke an expression, only an identifier), but when I use Operator.PostfixChainable it still throws.

Any idea what might be the cause for this issue?

Links to relevant code in my repo:
Parser: https://github.com/Drake53/War3Net/tree/master/src/War3Net.CodeAnalysis.Jass/Parser
Test project: https://github.com/Drake53/War3Net/tree/master/tests/War3Net.CodeAnalysis.Jass.Tests

**Comment by benjamin-hodgson** [comment]
Thanks for your interest! I'm having a hard time getting your code to build (VS is complaining about a missing project). Please could you try to reduce your code to a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)? It would make it much easier for me to help.

If I had to guess I'd say that [one of your term parsers](https://github.com/Drake53/War3Net/blob/925f6962060f9084b62e84f77a30cea8e5bd858c/src/War3Net.CodeAnalysis.Jass/Parser/JassPidginParser.cs#L27-L38) is accepting an empty string.

**Comment by Drake53** [comment]
I created a new repo for this issue: https://github.com/Drake53/JassPidginTest
It's not really minimal since I basically just copy-pasted the required files (including uncommited stuff that I've been working on since opening this issue).

The code now has two parsers, one that consumes a string, and another consuming syntax tokens. The latter reuses a tokenizer class which I created for an old version of my parser (before I started using your library), and doesn't fail any of the test cases.

I don't think any of my term parsers are able to parse an empty string, I created a test case to parse an empty string and both parsers throw as expected.
